lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -247,6 +247,20 @@ def tunit_edges(self, vals=None, M=None):
                  (tc[7], tc[4])]
         return edges
 
+    def apply_aspect(self, position=None):
+        if position is None:
+            position = self.get_position(original=True)
+
+        # in the superclass, we would go through and actually deal with axis
+        # scales and box/datalim. Those are all irrelevant - all we need to do
+        # is make sure our coordinate system is square.
+        figW, figH = self.get_figure().get_size_inches()
```

Comment:
Why use the figure's aspect ratio? What if the figure has 3 subplots all in a row?

Reply:
I think it makes sense.  I have no intention of spending the time required to understand mplot3d well enough to make a detailed review, but in the absence of any such review I am inclined to trust that this PR is a major improvement, as it appears to be fixing a basic design flaw.